dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/NeuralNetworkParameterUpdater.java

```diff
@@ -103,47 +100,24 @@ public NeuralNetParamServerData process(final String key,
     if (neuralNetParamServerData.isValidationStatsPair()) {
       return new NeuralNetParamServerData(neuralNetParamServerData.getValidationStatsPair());
     } else {
-      return new NeuralNetParamServerData(processLayerParametersList(key,
-          neuralNetParamServerData.getLayerParametersList()));
+      return new NeuralNetParamServerData(processLayerParameters(key, neuralNetParamServerData.getLayerParameters()));
     }
   }
 
   /**
-   * Aggregate parameter gradients by computing the average of all gradients, per layer,
-   * and multiplying the step size.
+   * Multiplies the step size (in-place update).
    */
-  private List<LayerParameter[]> processLayerParametersList(final String key,
-                                                            final List<LayerParameter[]> parameterGradientsList) {
-    if (parameterGradientsList == null || parameterGradientsList.size() == 0 || !key.equals(WHOLE_MODEL)) {
+  private LayerParameter[] processLayerParameters(final String key,
+                                                  final LayerParameter[] parameterGradients) {
+    if (parameterGradients == null || parameterGradients.length == 0 || !key.equals(WHOLE_MODEL)) {
```

Comment:
I'll remove a code for checking `parameterGradients.length == 0` at `processLayerParameters()` and add an error handling code for zero-length parameter gradients array at `updateLayerParameter()`.